### PR TITLE
Overhaul Contour creation

### PR DESCRIPTION
**Pull request by HendrikLeuschner**
### Pull Request Checklist

- [x] 1. I have [**rebased**](https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines) the latest version of the master branch into my feature branch and all conflicts have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the [Unreleased] heading.
- [x] 3. I have documented my code using JDocs tags.
- [x] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [x] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ x 6. I have created API tests for any new functionality exposed to the API.
- [x] 7. If changes/additions are made to the app.config file, I have added these to the app.config wiki page on github along with a short description of what it is for, and documented this in the Pull Request (below).
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [x] 10. For new features or changes involving building of graphs, I have tested on a larger dataset (at least Germany) and the graphs build without problems (i.e. no out-of-memory errors).
- [x] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the importer etc.), I have generated longer distance routes for the affected profiles with different options (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end points generated from the current live ORS. If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage and why the change was needed.

Fixes # .

### Information about the changes
- Key functionality added: Redid how border edge way geometry is handled in contour creation. Added buffering points for contour creation.
- Reason for change: Contours create out-of-cell "tails" from outgoing edges. This is a problem for ferry edges that are multiple km long. Also added buffering for contour creation, as otherwise singled out edges can create problematic geometries (see Borkum as example)

### Examples and reasons for differences between live ORS routes and those generated from this pull request
-

### Required changes to app.config (if applicable)
-

### CODE REVIEW

File: openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Eccentricity.java [hunk 1]
```diff
@@ -103,6 +105,7 @@ public void calcEccentricities(Weighting weighting, FlagEncoder flagEncoder) {
                 }
 
                 //TODO Maybe implement a logic smarter than having some high percentage for acceptedFullyReachable
+                int cellId = isochroneNodeStorage.getCellId(node);
```
Comment (takb) [inline openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Eccentricity.java]: What is this good for?

File: openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Contour.java [hunk 2]
```diff
@@ -17,6 +18,10 @@
 import org.heigit.ors.fastisochrones.partitioning.storage.CellStorage;
 import org.heigit.ors.fastisochrones.partitioning.storage.IsochroneNodeStorage;
 import org.heigit.ors.isochrones.builders.concaveballs.PointItemVisitor;
+import org.heigit.ors.routing.graphhopper.extensions.edgefilters.EdgeFilterSequence;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
```
Comment (takb) [inline openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Contour.java]: Unused imports?

File: openrouteservice/src/main/java/org/heigit/ors/fastisochrones/FastIsochroneAlgorithm.java [hunk 3]
```diff
@@ -184,6 +184,7 @@ public void approximateActiveCells(double approximation) {
             Map.Entry<Integer, IntObjectMap<SPTEntry>> activeCell = activeCellIterator.next();
             if (activeCell.getValue().size() / (double) cellStorage.getNodesOfCell(activeCell.getKey()).size() > approximation) {
                 activeCellIterator.remove();
+                activeCells.remove(activeCell.getKey());
```
Comment (takb) [inline openrouteservice/src/main/java/org/heigit/ors/fastisochrones/FastIsochroneAlgorithm.java]: A few lines above there is an unused local var, and you have an unused import and unused private method. Is your sonar working?

File: openrouteservice/src/main/java/org/heigit/ors/fastisochrones/partitioning/InertialFlow.java [hunk 4]
```diff
@@ -312,6 +325,10 @@ public void setInverseSemaphore(InverseSemaphore inverseSemaphore) {
         this.inverseSemaphore = inverseSemaphore;
     }
 
+    public void setFlagEncoder(FlagEncoder flagEncoder) {
+        this.flagEncoder = flagEncoder;
+    }
+
```
Comment (takb) [inline openrouteservice/src/main/java/org/heigit/ors/fastisochrones/partitioning/InertialFlow.java:331]: flagEncode is static, so should be 
```
public static void setFlagEncoder(FlagEncoder flagEncoder) {
    InertialFlow.flagEncoder = flagEncoder;
}
```

**Review by takb — CHANGES_REQUESTED** [review]
Code changes look ok, just some linting

File: openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Eccentricity.java (hunk 1 above)
Comment (HendrikLeuschner) [reply openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Eccentricity.java]: Removed

File: openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Contour.java (hunk 2 above)
Comment (HendrikLeuschner) [reply openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Contour.java]: Removed

File: openrouteservice/src/main/java/org/heigit/ors/fastisochrones/FastIsochroneAlgorithm.java (hunk 3 above)
Comment (HendrikLeuschner) [reply openrouteservice/src/main/java/org/heigit/ors/fastisochrones/FastIsochroneAlgorithm.java]: Removed

File: openrouteservice/src/main/java/org/heigit/ors/fastisochrones/partitioning/InertialFlow.java (hunk 4 above)
Comment (HendrikLeuschner) [reply openrouteservice/src/main/java/org/heigit/ors/fastisochrones/partitioning/InertialFlow.java:331]: Made static

**Comment by HendrikLeuschner** [comment]
I will rerun locally on Germany for verification again before merge.

**Review by MichaelsJP — APPROVED** [review]

**Review by takb — APPROVED** [review]

**Comment by HendrikLeuschner** [comment]
Runs fine for car on Germany. Also tested foot on Niedersachsen and it works fine as well. This can be merged.

**Pull request merged**